lmfit/parameter.py

```diff
@@ -238,19 +318,19 @@ def setup_bounds(self):
         The internal value for parameter from self.value (which holds
         the external, user-expected value).   This internal value should
         actually be used in a fit.
-       """
+        """
         if self.min in (None, -inf) and self.max in (None, inf):
             self.from_internal = lambda val: val
             _val  = self._val
         elif self.max in (None, inf):
-            self.from_internal = lambda val: self.min - 1 + sqrt(val*val + 1)
-            _val  = sqrt((self._val - self.min + 1)**2 - 1)
+            self.from_internal = lambda val: self.min - 1.0 + sqrt(val*val + 1)
+            _val  = sqrt((self._val - self.min + 1.0)**2 - 1)
         elif self.min in (None, -inf):
             self.from_internal = lambda val: self.max + 1 - sqrt(val*val + 1)
-            _val  = sqrt((self.max - self._val + 1)**2 - 1)
+            _val  = sqrt((self.max - self._val + 1.0)**2 - 1)
         else:
             self.from_internal = lambda val: self.min + (sin(val) + 1) * \
-                                 (self.max - self.min) / 2
+                                 (self.max - self.min) / 2.0
```

Comment:
We should have `from __future__ import division` to handle this kind of thing, part of the standard header for packages supporting 2 and 3.